zanata-war/src/main/java/org/zanata/adapter/PropertiesAdapter.java

```diff
@@ -0,0 +1,225 @@
+ * migrated to server
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Slf4j
+public class PropertiesAdapter implements FileFormatAdapter {
+
+    public static final String ISO_8859_1 = Charsets.ISO_8859_1.name();
+
+    public static final String UTF_8 = Charsets.UTF_8.name();
+
+    @Override
+    public Resource parseDocumentFile(URI fileUri, LocaleId sourceLocale,
+        Optional<String> filterParams) throws FileFormatAdapterException,
+        IllegalArgumentException {
+
+        return parseDocumentFile(fileUri, sourceLocale, filterParams,
+            ISO_8859_1);
+    }
+
+    @Override
+    public TranslationsResource parseTranslationFile(URI fileUri,
+        String localeId, Optional<String> params)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        return parseTranslationFile(fileUri, localeId, params, ISO_8859_1);
+    }
+
+    @Override
+    public void writeTranslatedFile(OutputStream output, URI originalFile,
+        Resource resource, TranslationsResource translationsResource,
+        String locale, Optional<String> params)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        writeTranslatedFile(output, resource, translationsResource, locale,
+                params, ISO_8859_1);
+    }
+
+    public Resource parseDocumentFile(URI fileUri, LocaleId sourceLocale,
+        Optional<String> filterParams, String charset) throws FileFormatAdapterException,
+        IllegalArgumentException {
+
+        if (sourceLocale == null) {
+            throw new IllegalArgumentException("Source locale cannot be null");
+        }
+
+        if (fileUri == null) {
+            throw new IllegalArgumentException("Document URI cannot be null");
+        }
+
+        PropReader propReader = new PropReader(charset, sourceLocale,
+            ContentState.Approved);
+
+        BufferedInputStream inputStream = null;
+        Resource doc = new Resource();
+
+        try {
+            inputStream = readStream(fileUri);
+            propReader.extractTemplate(doc, inputStream);
+        } catch (IOException e) {
+            throw new FileFormatAdapterException(
+                "Could not open the URL. The URL is OK but the input stream could not be opened.\n"
+                    + e.getMessage(), e);
+        } finally {
+            try {
+                inputStream.close();
+            } catch (IOException e1) {
+            }
+        }
+        return doc;
+    }
+
+    public TranslationsResource parseTranslationFile(URI fileUri,
+        String localeId, Optional<String> params, String charset)
+        throws FileFormatAdapterException, IllegalArgumentException {
+
+        if (StringUtils.isEmpty(localeId)) {
+            throw new IllegalArgumentException(
+                "locale id string cannot be null or empty");
+        }
+
+        PropReader propReader = new PropReader(charset, LocaleId.EN_US,
```

Comment:
> the en-us is source locale.

So if someone wants to use a Dutch properties source file, it will be impossible. Is that the correct implementation?